ksqldb-functional-tests/src/test/resources/query-validation-tests/greatest.json

```diff
@@ -0,0 +1,150 @@
+{
+  "comments": [
+    "Tests covering the greatest function."
+  ],
+  "tests": [
+    {
+      "name": "test greatest with numeric types",
+      "format": ["JSON"],
+      "statements": [
+        "CREATE STREAM INPUT (ID BIGINT KEY, N1 INT, N2 INT, N3 INT, N4 INT) WITH (kafka_topic='input_topic',value_format='{FORMAT}');",
+        "CREATE STREAM OUTPUT AS SELECT ID, GREATEST(N1, N2, N3, N4, null, null, 5) AS HIGHEST FROM INPUT;"
+      ],
+      "inputs": [
+        {"topic": "input_topic", "key": 1, "value": {"N1": 6, "N2": -1, "N3":  0, "N4": null}},
+        {"topic": "input_topic", "key": 2, "value": {"N1": 2147483647, "N2": -2147483648, "N3":  0, "N4": null}},
+        {"topic": "input_topic", "key": 3, "value": {"N1": null, "N2": null, "N3":  null, "N4": null}}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": 1, "value": {"HIGHEST": 6}},
+        {"topic": "OUTPUT", "key": 2, "value": {"HIGHEST": 2147483647}},
+        {"topic": "OUTPUT", "key": 3, "value": {"HIGHEST": 5}}
+      ]
+    },
+    {
+      "name": "test implicit casting to long",
+      "format": ["JSON"],
+      "statements": [
+        "CREATE STREAM INPUT (ID BIGINT KEY, N1 INT, N2 BIGINT) WITH (kafka_topic='input_topic',value_format='{FORMAT}');",
+        "CREATE STREAM OUTPUT AS SELECT ID, GREATEST(N1, N2, null) AS HIGHEST FROM INPUT;"
+      ],
+      "inputs": [
+        {"topic": "input_topic", "key": 1, "value": {"N1": 1, "N2": 1000000000000}}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": 1, "value": {"HIGHEST": 1000000000000}}
+      ]
+    },
```

Comment:
Let's add a schema check for the output (to test that the output column has the expected type), here and also in the tests below. Here's an example: https://github.com/confluentinc/ksql/blob/8f2d79944050af0402a208a9cd939eaf92d37851/ksqldb-functional-tests/src/test/resources/query-validation-tests/select.json#L20